### PR TITLE
feat: add a nox session to run application locally

### LINKED ISSUE #770: use nox to launch voila app in non-sepal env
I think it would be good if we can develop application outside of SEPAL and still rely on the venv that will be created. For this purpose I suggest to create a nox session to:
- create a venv using requirements.txt 
- create a temp jupyter kernel
- use the entr-point notebook
- launch it with the temp jupyter kernel 
- serve it on whenever localhost port available

To the developer that would loog like: 
```
nox -s app
```

and as long as the kernel would be running the nox env would be running as well

### LINKED ISSUE #771: cannot use modul_factory anymore
**Describe the bug**

When run I got an isort issue raised

### PR DESCRIPTION
**Pull request by 12rambau**
Fix #771, Fix #770 

Ok it's a 3 in 1 PR so sorry for the number of changed files, everything was interconnected. 

## apply D212

D212 state that docstring should not start with a line break. I don't like it but it's too complicated to enforce it everywhere (the template were in fact defaulting so I gave up and changed it everywhere in the lib (that 99% of the modifications)

## make factory work again

the `module_factory` command was broken because of isort moving there repository from gitlab to github. as isort is now included in ruff (sort of) I decided to update the whole pre-commit file for each template: 

- pre-commit file runs on ruff instead of isort/flake8 same version as sepal-ui
- inject parameters in a pyproject.toml file
- update the `main` which was defaulting to panel_app whatever the answer
- update `module_deploy` and `requirements.txt` to install more loose version of the main libs (related to https://github.com/openforis/sepal/pull/265)

# the crown jewel: run application locally in isolated env

that's the real deal: I installed a `noxfile.py` programm to run application from a local terminal with a isolated venv. got to any template, run `nox -s app` and the application should start on your local port. To end the process, close de tab and shutdown the terminal with "ctrl+c". 

the modifications: 
- update the pyproject.toml with the init notebook
- add a nox session to run application based on the requirements.txt and the init-notebook set in the pyproject.toml. The environment is isolated and voila is run in debug mode so you'll see errors when they are raised (they will be displayed in the terminal as well anyway). 
- this can easily be moved to any existing module as long as the requirements.txt are loose enough (my python version is so fa away from SEPAL that I cannot use any of the existing reqs)

@dfguerrerom  I'll let you run some test, let me know if you think it's a useful fonctionality and if yes I'll document it somewhere.

## TODO
- [ ] documentation

### CODE REVIEW

**Review by dfguerrerom — COMMENTED** [review]
I've checked it on my own environment and it works smoothly :100:, just a minor thing:
- I have installed `voila` but it's not part of apps requirements, so it would have failed if I hasn't.

### Edit:

I forgot to mention that docs build failed because of:

```python
Could not import extension sphinx-favicon (exception: No module named 'sphinx-favicon')

```

**Comment by 12rambau** [comment]
> Could not import extension sphinx-favicon (exception: No module named 'sphinx-favicon')

solved in https://github.com/12rambau/sepal_ui/pull/776

**Comment by 12rambau** [comment]
I think I solved the import issues for: mypy, toml, and voila
You just need to reinstall the dev version of sepal_ui

File: sepal_ui/bin/module_deploy.py [hunk 1]
```diff
@@ -203,9 +195,11 @@ def main() -> None:
     print("Export the env configuration of your module...")
 
     # check that the local folder is a module folder
-    ui_file = Path.cwd() / "ui.ipynb"
-    if not ui_file.is_file():
-        raise Exception(f"{Fore.RED}This is not a module folder.")
+    toml = Path.cwd() / "pyproject.toml"
+    try:
+        toml.load("pyproject.toml")["sepal-ui"]["init-notebook"]
+    except Exception as e:
```
[the remark this answers is not in the document]
Comment (12rambau) [reply sepal_ui/bin/module_deploy.py]: fix in https://github.com/12rambau/sepal_ui/pull/775/commits/fe6c885320cbc8cbd24c10eb11913f6f8ec22e81

File: sepal_ui/templates/panel_app/noxfile.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+"""All the process that can be run using nox.
+
+The nox run are build in isolated environment that will be stored in .nox. to force the venv update, remove the .nox/xxx folder.
+"""
+
+import nox
+import toml
```
[the remark this answers is not in the document]
Comment (12rambau) [reply sepal_ui/templates/panel_app/noxfile.py:7]: fixed in https://github.com/12rambau/sepal_ui/pull/775/commits/54051607798ddf5effc07d06caae0d429ee5d4af

**Pull request merged**